inc/sitemaps/class-sitemaps.php

```diff
@@ -0,0 +1,436 @@
+<?php
+/**
+ * @package WPSEO\XML_Sitemaps
+ */
+
+/**
+ * Class WPSEO_Sitemaps
+ */
+class WPSEO_Sitemaps {
+
+	/** @var string $sitemap Content of the sitemap to output. */
+	protected $sitemap = '';
+
+	/** @var string $stylesheet XSL stylesheet for styling a sitemap for web browsers. */
+	private $stylesheet = '';
+
+	/** @var bool $bad_sitemap Flag to indicate if this is an invalid or empty sitemap. */
+	public $bad_sitemap = false;
+
+	/** @var bool $transient Whether or not the XML sitemap was served from a transient or not. */
+	private $transient = false;
+
+	/** @var int $max_entries The maximum number of entries per sitemap page. */
+	private $max_entries;
+
+	/** @var string $http_protocol HTTP protocol to use in headers. */
+	protected $http_protocol = 'HTTP/1.1';
+
+	/** @var int $current_page Holds the n variable. */
+	private $current_page = 1;
+
+	/** @var WPSEO_Sitemap_Timezone $timezone */
+	private $timezone;
+
+	/** @var WPSEO_Sitemaps_Router $router */
+	public $router;
+
+	/** @var WPSEO_Sitemaps_Renderer $renderer  */
+	public $renderer;
+
+	/** @var WPSEO_Sitemap_Provider[] $providers */
+	public $providers;
+
+	/**
+	 * Class constructor
+	 */
+	public function __construct() {
+
+		if ( ! defined( 'ENT_XML1' ) ) {
```

Comment:
I have no idea. :D It's used in some character encoding stuff later. Will look it up and document in stage 3.